presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -212,12 +216,24 @@ public void abort()
 
     private void doAbort()
     {
+        Map<HiveWriter, Exception> exceptions = new HashMap<>();
         for (HiveWriter writer : writers) {
             // writers can contain nulls if an exception is thrown when doAppend expends the writer list
             if (writer != null) {
-                writer.rollback();
+                try {
+                    writer.rollback();
+                }
+                catch (Exception e) {
+                    exceptions.put(writer, e);
+                }
             }
         }
+        if (!exceptions.isEmpty()) {
+            exceptions.entrySet().forEach(
+                    entry -> log.warn("exception '%s' while rollback on %s", entry.getValue(), entry.getKey())
```

Comment:
Why not print the logs as you encounter that? The list doesn't appear to be necessary.